### PR TITLE
Implement WemoDiscoveryService and remove dependency of JUPnP to make…

### PR DESCRIPTION
**Pull request by hmerk**
… Wemo Binding work with the new firmware versions.

Signed-off-by: hmerk <hans-joerg.merk@t-online.de>

### CODE REVIEW

File: addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.binding.wemo.handler.WemoHandler;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
```
Comment (kaikreuzer) [inline addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java]: you are lacking an according import package in the MANIFEST.MF

File: addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+		super(WemoHandler.SUPPORTED_THING_TYPES, 15, true);
+	}
+ 
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return WemoHandler.SUPPORTED_THING_TYPES;
+    }
+ 
+	protected void startScan() {
+		logger.debug("Start WeMo Device discovery");
+		scheduler.scheduleAtFixedRate(wemoDiscoveryRunnable, 0, refreshInterval, TimeUnit.SECONDS);
```
Comment (kaikreuzer) [inline addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java]: startScan should actually only trigger it exactly once!

File: addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+				logger.debug("Send datagram packet.");
+				wemoReceiveSocket.send(discoveryPacket);
+
+				while (true) {
+					try {
+						logger.debug("Receive SSDP Message.");
+						receivePacket = new DatagramPacket(new byte[1536], 1536);
+						wemoReceiveSocket.receive(receivePacket);
+						final String message = new String(receivePacket.getData());
+						logger.debug("Recieved message: {}", message);
```
Comment (kaikreuzer) [inline addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java]: typo: Received

File: addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+				multicast.close();
+			}
+
+			// Response-Listener
+			DatagramSocket wemoReceiveSocket = null;
+			DatagramPacket receivePacket = null;
+			try {
+				wemoReceiveSocket = new DatagramSocket(SSDP_SEARCH_PORT);
+				wemoReceiveSocket.setSoTimeout(TIMEOUT);
+				logger.debug("Send datagram packet.");
```
Comment (kaikreuzer) [inline addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java]: probably worth to change a few log statements to trace.

File: addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+												logger.debug("Wemo device found at URL '{}'", wemoLocation);
+												wemoFriendlyName = StringUtils.substringBetween(response, "<friendlyName>", "</friendlyName>");
+												logger.debug("Wemo device '{}' found at '{}'", wemoFriendlyName, wemoLocation);
+												wemoModelName = StringUtils.substringBetween(response, "<modelName>", "</modelName>");
+												logger.debug("Wemo device '{}' has model name '{}'", wemoFriendlyName, wemoModelName);
+												label = "Wemo" + wemoModelName;
+												
+												switch(wemoModelName) {
+												case "Socket":
+													logger.debug("Cerating ThingUID for device model '{}' with UDN '{}'", wemoModelName, wemoUDN);
```
Comment (kaikreuzer) [inline addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java]: typo: Creating
Also in the following lines

File: addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+												case "LightSwitch":
+													logger.debug("Cerating ThingUID for device model '{}' with UDN '{}'", wemoModelName, wemoUDN);
+													uid = new ThingUID(WEMO_LIGHTSWITCH_TYPE_UID, wemoUDN);
+													break;
+												case "Motion":
+													logger.debug("Cerating ThingUID for device model '{}' with UDN '{}'", wemoModelName, wemoUDN);
+													uid = new ThingUID(WEMO_MOTION_TYPE_UID, wemoUDN);
+													break;
+												}
+/**
```
Comment (kaikreuzer) [inline addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java]: please remove

File: addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+													logger.debug("Cerating ThingUID for device model '{}' with UDN '{}'", wemoModelName, wemoUDN);
+													uid = new ThingUID(WEMO_LIGHTSWITCH_TYPE_UID, wemoUDN);
+												}
+												if (wemoModelName == "Motion") {
+													logger.debug("Cerating ThingUID for device model '{}' with UDN '{}'", wemoModelName, wemoUDN);
+													uid = new ThingUID(WEMO_MOTION_TYPE_UID, wemoUDN);
+												}
+*/											
+									            Map<String, Object> properties = new HashMap<>(4);
+									            properties.put("label", label);
```
Comment (kaikreuzer) [inline addons/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/discovery/WemoDiscoveryService.java]: what is this good for?

**Comment by kaikreuzer** [comment]
Thanks, seems to work smoothly! Please incorporate my few comments and squash your commits - thanks!

**Comment by hmerk** [comment]
Hmmm, I think I messed up my lokal fork an now I am not able to squash the two commits.
Any hint for me ??

**Comment by kaikreuzer** [comment]
I wouldn't mind to accept the PR with the two commits inside as well. But I just have another comment: Although the image in the documentation looks very nice, I doubt that you have the rights to contribute it, right? So please either remove it or use a picture where you have the rights (e.g. you took it yourself).

**Pull request closed without merging**